### PR TITLE
Handle filetype field in kubernetes.yaml files

### PR DESCRIPTION
**Pull request by rhatdan**
Signed-off-by: Daniel J Walsh <dwalsh@redhat.com>

#### Does this PR introduce a user-facing change?

```release-note
podman kube generate and play now support SELinux filetype labels.
```

### CODE REVIEW

File: test/system/helpers.bash [hunk 1]
```diff
@@ -347,6 +347,11 @@ function is_aarch64() {
     [ "$(uname -m)" == "aarch64" ]
 }
 
+function is_selinux() {
+    /usr/sbin/selinuxenabled 2> /dev/null
+    return $?
```
Comment (edsantiago) [inline test/system/helpers.bash:352]: Unnecessary; please remove, it's misleading.

File: test/system/700-play.bats [hunk 2]
```diff
@@ -549,3 +549,18 @@ EOF
...
+    run cat $YAML
+    is "$output" ".*filetype: usr_t" "Generated YAML file should container filetype usr_t"
+    run_podman pod rm --force pod1
+
+    run_podman kube play $YAML
+    if is_selinux; then
+        run_podman inspect pod1-test1 --format "{{ .MountLabel }}"
+	is "$output" "system_u:object_r:usr_t:s0:c1,c2" "Generated container should use filetype usr_t"
+    fi
+}
```
Comment (edsantiago) [inline test/system/700-play.bats:566]: There's probably some cleanup missing here?

File: test/system/helpers.bash [hunk 3]
```diff
@@ -347,6 +347,11 @@ function is_aarch64() {
     [ "$(uname -m)" == "aarch64" ]
 }
 
+function is_selinux() {
```
Comment (edsantiago) [inline test/system/helpers.bash:350]: I mean, you're the expert, but wouldn't `selinux_enabled` be a better name?

**Comment by edsantiago** [comment]
tests LGTM, I can't review the code though

File: test/system/700-play.bats [hunk 4]
```diff
@@ -549,3 +549,19 @@ EOF
 
     run_podman kube down $yaml_source
 }
+
+@test "podman kube generate filetype" {
+    YAML=$PODMAN_TMPDIR/test.yml
+    run_podman create --pod new:pod1 --security-opt label=level:s0:c1,c2 --security-opt label=filetype:usr_t --name test1 $IMAGE true
+    run_podman kube generate pod1 -f $YAML
+    run cat $YAML
+    is "$output" ".*filetype: usr_t" "Generated YAML file should container filetype usr_t"
```
Comment (vrothberg) [inline test/system/700-play.bats:558]:
```suggestion
    is "$output" ".*filetype: usr_t" "Generated YAML file should contain filetype usr_t"
```

File: test/system/700-play.bats (hunk 2 above)
Comment (vrothberg) [reply test/system/700-play.bats:567]: Yes, it looks like the initial pod/container used in `kube generate` need to be removed as well.

Comment (rhatdan) [reply test/system/700-play.bats:567]: run_podman pod rm --force pod1 
Should remove it.

File: test/system/700-play.bats (hunk 4 above)
Comment (rhatdan) [reply test/system/700-play.bats]: My fingers automatically add the er whenever I type contain ...

**Pull request merged**